.gitlab-ci.yml

```diff
@@ -526,8 +531,13 @@ release_circleci_runner_gcr:
   rules:
```

Comment:
Note: with this change, `dev_release_circleci_runner_gcr` and `release_circleci_runner_gcr` are identical. That's strange, we should check why that is the case.

Reply:
It *seems* that the intent was to somehow replicate the '${TEST_ONLY}` behavior, by having one job for the default branch and one job for other pushes (also including `main` FWICS)
This can probably be merged in a single job, but I'd prefer to do so once we're back to a working state (and quite possibly with more thorough verifications)